src/simulation/deviceInterface/tempMeasurement/tempMeasurement.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+ ISC License
+
+ Copyright (c) 2023, Autonomous Vehicle Systems Lab, University of Colorado Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+*/
+
+
+#include "tempMeasurement.h"
+#include <iostream>
+#include <cstring>
+
+/*! This is the constructor for the module class.  It sets default variable
+    values and initializes the various parts of the model */
+TempMeasurement::TempMeasurement()
+{
+    this->trueTemperature = 0.0;
+    this->sensedTemperature = 0.0;
+
+    // Initialize fault parameers
+    this->senNoiseStd = 0.0;
+    this->senBias = 0.0;
+    this->walkBounds = 1E-15; //don't allow random walk by default
+    this->noiseModel = GaussMarkov(1, this->RNGSeed);
+    this->faultState = TEMP_FAULT_NOMINAL;
+    this->pastValue = 0.0;
+    this->stuckValue = 0.0;
+    this->spikeProbability = 0.1; // default 10% chance of spiking if in fault state
+    this->spikeAmount = 2.0; // default to double the value if spiking
+}
+
+/*! Module Destructor */
+TempMeasurement::~TempMeasurement()
```

Comment:
:small_red_triangle_down: You can set this destructor to default as follows. The reason is explained at the below link, but TLDR; doing so saves us from messing up. Make it a habit, and a custom destructor the exception to the habit. That way when you need to get it right you'll know, because you'll necessarily be writing a custom destructor. 

```TempMeasurement::~TempMeasurement() = default;```

https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#c30-define-a-destructor-if-a-class-needs-an-explicit-action-at-object-destruction

Reply:
We should fix this, and other comments you've made about deleting comments etc., in the makeDraftModule.py code that autogenerates new modules.

Reply:
Done.